src/server/routes/plywood/plywood.ts

```diff
@@ -17,16 +17,28 @@
 
 import { Timezone } from "chronoshift";
 import { Request, Response, Router } from "express";
-import { Dataset, Expression } from "plywood";
+import { IncomingHttpHeaders } from "http";
+import { Dataset, Expression, RefExpression } from "plywood";
+import { DynamicSubsetFormula } from "../../../common/models/dynamic-subset-formula/dynamic-subset-formula";
 import { checkAccess } from "../../utils/datacube-guard/datacube-guard";
 import { SettingsGetter } from "../../utils/settings-manager/settings-manager";
 
+function applySubset(expression: Expression, dynamicSubsetFormula: DynamicSubsetFormula, headers: IncomingHttpHeaders): Expression {
+  const subsetFilter = dynamicSubsetFormula.fn(headers);
```

Comment:
Add some checks:

- does `dynamicSubsetFormula` exists?
- does it throw?
- is returned value a valid Expression?

Reply:
done